sys/include/net/gnrc/icmpv6/echo.h

```diff
@@ -58,6 +59,53 @@ gnrc_pktsnip_t *gnrc_icmpv6_echo_build(uint8_t type, uint16_t id, uint16_t seq,
 void gnrc_icmpv6_echo_req_handle(gnrc_netif_t *netif, ipv6_hdr_t *ipv6_hdr,
                                  icmpv6_echo_t *echo, uint16_t len);
 
+/**
+ * @brief   Send out ICMPv6 echo request
+ *
+ * @param[in] netif     The interface the echo request should be sent on.
+ * @param[in] addr      The destination address of the echo request
+ * @param[in] id        ID for the echo message in host byte-order
+ * @param[in] seq       Sequence number for the echo message in host byte-order
+ * @param[in] ttl       Hop limit of the echo request
+ * @param[in] len       Length of the payload
+ *
+ * @return  0 on success
+ * @return <0 on error
+ */
+int gnrc_icmpv6_echo_send(const gnrc_netif_t *netif, const ipv6_addr_t *addr,
+                          uint16_t id, uint16_t seq, uint8_t ttl, size_t len);
+
+/**
+ * @brief   ICMPv6 echo response callback
+ *
+ * @param[in] icmpv6    ICMPv6 header
+ * @param[in] from      The source address of the echo response
+ * @param[in] hoplimit  Hop limit of the echo response
+ * @param[in] netif_hdr l2 header (may be NULL)
+ * @param[in] corrupt   Offset of corrupt payload, -1 if no corruption detected
+ * @param[in] rtt_us    round-trip-time in µs (if available)
+ * @param[in] ctx       User supplied context
+ *
+ * @return  0 on success
+ * @return <0 on error
+ */
+typedef int (*gnrc_icmpv6_echo_rsp_handle_cb_t)(gnrc_pktsnip_t *icmpv6, ipv6_addr_t *from,
+                                                uint8_t hoplimit, gnrc_netif_hdr_t *netif_hdr,
+                                                int corrupt, uint32_t rtt_us, void *ctx);
```

Comment:
Do we really need to separate `icmpv6`, from, and `netif_hdr` here? The packet containing both `icmpv6` and `netif_hdr` should be enough.